coalib/coala_main.py

```diff
@@ -104,27 +103,27 @@ def run_coala(console_printer=None,
         while True:
             for i, action in enumerate(all_actions_possible, 1):
                 console_printer.print(format_lines('{}'.format(
-                    action), symbol='['))
+                    action), symbol='[', line_nr=' ' + str(i-1) + '.'))
 
             line = format_lines(STR_ENTER_NUMBER, symbol='[')
 
             choice = input(line)
 
-            if choice.isalpha():
+            if choice.isalnum():
                 choice = choice.upper()
-                choice = '(' + choice + ')'
-                if choice == '(N)':
+                choice = '(' + choice + ')' if choice.isalpha() else choice
+                if choice == '(N)' or choice == '0':
                     apply_single = 'Do (N)othing'
                     break
                 for i, action in enumerate(all_actions_possible, 1):
-                    if choice in action:
+                    if choice in action or choice in str(i-1):
                         apply_single = action
                         break
+
                 if apply_single:
                     break
-                console_printer.print(format_lines(
-                                    'Please enter a valid letter.',
-                                    symbol='['))
+            console_printer.print(format_lines(
```

Comment:
this doesn't seem correct

Reply:
May i know why? When the user doesn't enter a number or a alphabet it comes out of the if choice.isaplha() and prints the error message.

Reply:
won't this be printed either way whether the the condition is true or false?

Reply:
It seems strange, even i thought that. But it doen't print the error message when a correct option is given.

Reply:
The output looks like this :-
coala --bears PEP8Bear,SpaceConsistencyBear,PyCommentedBear,PycodestyleBear --files coala_main.py -A
[    ] 0. Do (N)othing
[    ] 1. (S)how patch
[    ] 2. (A)pply patch
[    ] 3. Add (I)gnore comment
[    ] 4. (O)pen file
[    ] 5. Print Aspec(T) Information
[    ] 6. Print (D)ebug message
[    ] 7. Print (M)ore info
[    ] Enter number (Ctrl-D to exit): 34
[    ] Please enter a valid letter or number.
[    ] 0. Do (N)othing
[    ] 1. (S)how patch
[    ] 2. (A)pply patch
[    ] 3. Add (I)gnore comment
[    ] 4. (O)pen file
[    ] 5. Print Aspec(T) Information
[    ] 6. Print (D)ebug message
[    ] 7. Print (M)ore info
[    ] Enter number (Ctrl-D to exit): 1
[WARNING][13:21:01] Default coafile '.coafile' not found!
Here's what you can do:
* add `--save` to generate a config file with your current options
* add `-I` to suppress any use of config files

[WARNING][13:21:01] No bears matching 'PyCommentedBear' were found. Make sure you have coala-bears installed or you have typed the name correctly.
[WARNING][13:21:01] acquire_settings: section parameter is deprecated.
Please enter a value for the setting "use_spaces" (True if spaces are to be used instead of tabs.) needed by SpaceConsistencyBear for section "cli": 
True
Executing section cli...

coala_main.py
[   9] from·coalib.processes.Processing·import·execute_section,·simplify_section_result
**** PycodestyleBear (E501) [Section: cli | Severity: NORMAL] ****
!    ! E501 line too long (80 > 79 characters)

Reply:
@ishanSrt  It won't print the console.print statement every time because, we are checking for
" if apply_single: break" before printing the error message. And in case the user enters the correct option apply_single becomes true and it breaks from the while loop. If the user doen't then that if conditon fails and it comes to the error message part.